Apps/WebClient/src/ClientApp/src/components/calendar/header.vue

```diff
@@ -101,9 +101,9 @@ export default class CalendarComponent extends Vue {
     }
 
     private dispatchEvent() {
-        this.title = moment(this.currentDate).format(this.titleFormat);
         let startDate = DateUtil.getMonthFirstDate(this.headerDate);
         this.$emit("update:currentDate", startDate);
+        this.title = moment(startDate).format(this.titleFormat);
```

Comment:
Instead of storing the title we should calculate it from the date
```
@Prop() currentDate!: Date;
    @Prop() titleFormat!: string;

    private headerDate: Date = new Date();
    private leftIcon: string = "chevron-left";
    private rightIcon: string = "chevron-right";

    public get title(): string {
        return moment(this.currentDate).format(this.titleFormat);
    }
```

Reply:
code updated and tested per @FuriousLlama 's suggestion